tests/test_recorder.py

```diff
@@ -99,19 +100,26 @@ def test_put_annotation_metadata():
     assert not subsegment.metadata['default'].get('key1')
 
 
-def test_pass_through_with_missing_context():
+def test_default_pass_through_with_missing_context():
     xray_recorder = get_new_stubbed_recorder()
-    xray_recorder.configure(sampling=False, context_missing='LOG_ERROR')
+    xray_recorder.configure(sampling=False) # default context_missing = 'LOG_ERROR'
     assert not xray_recorder.is_sampled()
 
     xray_recorder.put_annotation('key', 'value')
     xray_recorder.put_metadata('key', 'value')
     xray_recorder.end_segment()
 
+def test_raise_runtime_error_with_missing_context():
+    xray_recorder = get_new_stubbed_recorder()
+    xray_recorder.configure(sampling=False, context_missing='RUNTIME_ERROR')
+
+    with pytest.raises(SegmentNotFoundException):
```

Comment:
I'm not too familiar with Python. If there is not an exception throw, as expected, then will this test fail?

Reply:
It will! This is what happens when you modify the `context_missing` parameter to be `LOG_ERROR` in this test:
`Failed: DID NOT RAISE <class 'aws_xray_sdk.core.exceptions.exceptions.SegmentNotFoundException'>`

Reply:
Adding to Carol's comment above, this is a common way in pytest to assert that a certain exception is thrown in a unit test. Documentation for reference: https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions